sale_automatic_workflow_payment_mode/models/automatic_workflow_job.py

```diff
@@ -0,0 +1,49 @@
+# -*- coding: utf-8 -*-
+# © 2016 Camptocamp SA, Sodexis
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+import logging
+from openerp import models, api, fields
+from openerp.tools.safe_eval import safe_eval
+from openerp.addons.sale_automatic_workflow.models.automatic_workflow_job \
+    import commit
+
+_logger = logging.getLogger(__name__)
+
+
+class AutomaticWorkflowJob(models.Model):
+    _inherit = 'automatic.workflow.job'
+
+    @api.model
+    def run_with_workflow(self, sale_wkf):
+        workflow_domain = [('workflow_process_id', '=', sale_wkf.id)]
+        res = super(AutomaticWorkflowJob, self).run_with_workflow(sale_wkf)
+        if sale_wkf.register_payment:
+            self._register_payments(
+                safe_eval(sale_wkf.payment_filter_id.domain) +
+                workflow_domain)
+        return res
+
+    @api.model
+    def _register_payments(self, payment_filter):
+        invoice_obj = self.env['account.invoice']
+        invoices = invoice_obj.search(payment_filter)
+        _logger.debug('Invoices to Register Payment: %s', invoices.ids)
+        for invoice in invoices:
+            partner_type = invoice.type in ('out_invoice', 'out_refund') and \
+                'customer' or 'supplier'
+            payment_mode = invoice.payment_mode_id
+            with commit(self.env.cr):
```

Comment:
This is how `sale_automatic_workflow` works, every individual action is directly committed so even if something fails, the rest stays.